pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go

```diff
@@ -134,7 +127,8 @@ func (w *Watcher) traversePluginDir(dir string) error {
 		}
 
 		switch mode := info.Mode(); {
-		case mode.IsDir():
+		// do not call fsWatcher.Add twice on the root dir to avoid potential problems.
+		case path != dir && mode.IsDir():
```

Comment:
keep the case as-is so we don't fall through to the default case:

```suggestion
		case mode.IsDir():
			if path != dir {
				if err := w.fsWatcher.Add(path); err != nil {
					return fmt.Errorf("failed to watch %s, err: %v", path, err)
				}
			}
```